kotlin/navigate-route/src/main/java/com/esri/arcgisruntime/sample/navigateroute/MainActivity.kt

```diff
@@ -0,0 +1,305 @@
+/*
+ *  Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.esri.arcgisruntime.sample.navigateroute
+
+import android.content.res.Resources
+import android.graphics.Color
+import android.os.Build
+import android.os.Bundle
+import android.speech.tts.TextToSpeech
+import android.text.format.DateUtils
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.mapping.view.LocationDisplay
+import com.esri.arcgisruntime.navigation.DestinationStatus
+import com.esri.arcgisruntime.navigation.RouteTracker
+import com.esri.arcgisruntime.navigation.TrackingStatus
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteParameters
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteResult
+import com.esri.arcgisruntime.tasks.networkanalysis.RouteTask
+import com.esri.arcgisruntime.tasks.networkanalysis.Stop
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.layout_navigation_controls.*
+import java.util.concurrent.ExecutionException
+
+class MainActivity : AppCompatActivity() {
+
+  private var textToSpeech: TextToSpeech? = null
+
+  private var isTextToSpeechInitialized = false
+
+  companion object {
+    private val TAG: String = this::class.java.simpleName
+
+    /**
+     * Creates a list of stops along a route.
+     */
+    private fun getStops(): List<Stop> {
```

Comment:
maybe stops could be an object and just .apply { each point }

Reply:
I'd also put this below onCreate

Reply:
I'm not sure what you mean here. For now Ive moved it out to a package-level val which just constructs a list, which looks cleaner.